diff --git a/src/widgets/kernel/qwidgetwindow.cpp b/src/widgets/kernel/qwidgetwindow.cpp
index 279c6c0..991a05f 100644
--- a/src/widgets/kernel/qwidgetwindow.cpp
+++ b/src/widgets/kernel/qwidgetwindow.cpp
@@ -899,10 +899,10 @@
         const QPoint mapped = widget->mapFromGlobal(m_widget->mapToGlobal(event->pos()));
         QDragMoveEvent translated(mapped, event->possibleActions(), event->mimeData(),
                                   event->mouseButtons(), event->keyboardModifiers());
-        translated.setDropAction(event->dropAction());
-        translated.setAccepted(event->isAccepted());
 
         if (widget == m_dragTarget) { // Target widget unchanged: Send DragMove
+            translated.setDropAction(event->dropAction());
+            translated.setAccepted(event->isAccepted());
             QGuiApplication::forwardEvent(m_dragTarget, &translated, event);
         } else {
             if (m_dragTarget) { // Send DragLeave to previous
@@ -912,6 +912,9 @@
             }
             // Send DragEnter to new widget.
             handleDragEnterEvent(static_cast<QDragEnterEvent*>(event), widget);
+            // Handling 'DragEnter' should suffice for the application.
+            translated.setDropAction(event->dropAction());
+            translated.setAccepted(event->isAccepted());
             // The drag enter event is always immediately followed by a drag move event,
             // see QDragEnterEvent documentation.
             QGuiApplication::forwardEvent(m_dragTarget, &translated, event);
