diff -ur a/qtbase/mkspecs/common/icc-base-unix.conf b/qtbase/mkspecs/common/icc-base-unix.conf
--- a/qtbase/mkspecs/common/icc-base-unix.conf	2020-07-14 14:28:26.623839096 +0200
+++ b/qtbase/mkspecs/common/icc-base-unix.conf	2020-07-16 08:32:54.352133540 +0200
@@ -34,6 +34,9 @@
 QMAKE_CFLAGS_LTCG       = -ipo
 QMAKE_CFLAGS_DISABLE_LTCG = -no-ipo
 
+QMAKE_CFLAGS_C99       = -std=c99
+QMAKE_CFLAGS_C11       = -std=c11
+
 QMAKE_CFLAGS_SSE2      += -msse2
 QMAKE_CFLAGS_SSE3      += -msse3
 QMAKE_CFLAGS_SSSE3     += -mssse3
diff -ur a/qtbase/src/corelib/global/qnumeric_p.h b/qtbase/src/corelib/global/qnumeric_p.h
--- a/qtbase/src/corelib/global/qnumeric_p.h	2020-07-14 14:28:27.654831124 +0200
+++ b/qtbase/src/corelib/global/qnumeric_p.h	2020-07-16 08:33:46.691749151 +0200
@@ -224,7 +224,7 @@
 // size_t. Implementations for 8- and 16-bit types will work but may not be as
 // efficient. Implementations for 64-bit may be missing on 32-bit platforms.
 
-#if (defined(Q_CC_GNU) && (Q_CC_GNU >= 500) || (defined(Q_CC_INTEL) && !defined(Q_OS_WIN))) || QT_HAS_BUILTIN(__builtin_add_overflowx)
+#if (defined(Q_CC_GNU) && (Q_CC_GNU >= 500) || (defined(Q_CC_INTEL) && (Q_CC_GNU >= 500) && !defined(Q_OS_WIN))) || QT_HAS_BUILTIN(__builtin_add_overflowx)
 // GCC 5, ICC 18, and Clang 3.8 have builtins to detect overflows
 
 template <typename T> inline
