diff -ur a/src/core/web_engine_library_info.cpp b/src/core/web_engine_library_info.cpp
--- a/src/core/web_engine_library_info.cpp	2019-12-08 09:18:34.000000000 +0100
+++ b/src/core/web_engine_library_info.cpp	2020-09-02 11:14:48.125754392 +0200
@@ -259,15 +259,12 @@
     if (!initialized) {
         initialized = true;
         if (!QFileInfo::exists(potentialResourcesPath % QLatin1String("/icudtl.dat"))) {
-            qWarning("Qt WebEngine ICU data not found at %s. Trying parent directory...", qPrintable(potentialResourcesPath));
             potentialResourcesPath = QLibraryInfo::location(QLibraryInfo::DataPath);
         }
         if (!QFileInfo::exists(potentialResourcesPath % QLatin1String("/icudtl.dat"))) {
-            qWarning("Qt WebEngine ICU data not found at %s. Trying application directory...", qPrintable(potentialResourcesPath));
             potentialResourcesPath = QCoreApplication::applicationDirPath();
         }
         if (!QFileInfo::exists(potentialResourcesPath % QLatin1String("/icudtl.dat"))) {
-            qWarning("Qt WebEngine ICU data not found at %s. Trying fallback directory... The application MAY NOT work.", qPrintable(potentialResourcesPath));
             potentialResourcesPath = fallbackDir();
         }
     }
